### PR TITLE
Changes I made for the http-api-gateway component

### PR DESCRIPTION
**Pull request by cap10morgan**
These are the changes I made to the API for the http-api-gateway component (rename coming shortly): https://github.com/fluree/http-api-server

Happy to change anything in here, just let me know.

Part of #267

### CODE REVIEW

**Review by bplatz — APPROVED** [review]
I think this all looks good!

File: src/fluree/db/json_ld/api.cljc [hunk 1]
```diff
@@ -122,6 +123,34 @@
...
+(defn load-if-exists
+  "Attempts to load ledger-alias, returning a promise containing nil if it
+  doesn't exist, the ledger otherwise."
+  [conn ledger-alias]
+  (promise-wrap
+    (go
+      (let [address (<! (alias->address conn ledger-alias))
+            _ (log/debug "Attempting to load ledger from" address)
+            attempt (<! (jld-ledger/load conn address))]
+        (when-not (util/exception? attempt)
```
Comment (zonotope) [inline src/fluree/db/json_ld/api.cljc:151]: Minor nitpick, but there could be an exception here from a different error other than the ledger not existing. In that case the issue would be swallowed and the user would assume the ledger doesn't exist when it does but it just has another problem. 

I don't think it's a big deal for now, but maybe we could look for a specific exception to ignore, and pass everything else through.

**Review by zonotope — APPROVED** [review]
This looks fine to me, but I do have a minor suggestion that you can take or leave as you see fit.

File: src/fluree/db/json_ld/api.cljc [hunk 2]
```diff
@@ -122,6 +123,34 @@
...
+  "Loads an existing ledger by its alias (which will be converted to a
+  connection-specific address first)."
+  [conn ledger-alias]
+  (promise-wrap
+    (go
+      (let [address (<! (alias->address conn ledger-alias))]
+        (log/debug "Loading ledger from" address)
+        (<! (jld-ledger/load conn address))))))
+
+(defn load-if-exists
```
Comment (dpetran) [inline src/fluree/db/json_ld/api.cljc:142]: Are all these `load`* functions meant to be part of the public api? It's not clear to me when I would use `load` vs `load-if-exists`.

Comment (cap10morgan) [reply src/fluree/db/json_ld/api.cljc:142]: Yeah, there's definitely room for improvement here. Basically I needed a way to implement idempotent `create`. This was my "simplest possible thing that could possibly work" but I'd be happy to push it a little further in the direction of the ideal.

File: src/fluree/db/json_ld/api.cljc (hunk 1 above)
Comment (cap10morgan) [reply src/fluree/db/json_ld/api.cljc:151]: True, true. I'll investigate. And/or eliminate this fn entirely. :)

**Comment by cap10morgan** [comment]
OK @bplatz, @zonotope, and @dpetran I added an `exists?` fn to the API (and supporting protocol stuff and some tests for i it too that even run and pass in CLJS). If this all looks OK I'll adapt the http-api-server/gateway to use it.

**Comment by bplatz** [comment]
Sounds good! Although as it relates to API and JS compatibility the `?` is a bit troublesome. I think you'll either have to rename the API for just those, or consider renaming it for everything so it is consistent.

**Comment by cap10morgan** [comment]
> Sounds good! Although as it relates to API and JS compatibility the `?` is a bit troublesome. I think you'll either have to rename the API for just those, or consider renaming it for everything so it is consistent.

I was going to export it as `jldExists` in flureedb.cljs and flureenjs.cljs. Does that solve the (whole) problem?

**Comment by bplatz** [comment]
We only did the `jld` prefix originally to decide how we were going to reconcile the existing API and the JLD-specific calls.

Since we ditched backwards compatibility, all of the `jld` prefixes should get removed as this is now _the_ API.

We should probably have a ticket opened for that and get it into alpha 1 so we don't break API after release.

**Comment by cap10morgan** [comment]
> We only did the `jld` prefix originally to decide how we were going to reconcile the existing API and the JLD-specific calls.
> 
> Since we ditched backwards compatibility, all of the `jld` prefixes should get removed as this is now _the_ API.
> 
> We should probably have a ticket opened for that and get it into alpha 1 so we don't break API after release.

Ah, sure. In that case it would just be named `exists` in those namespaces.

**Comment by cap10morgan** [comment]
> We only did the `jld` prefix originally to decide how we were going to reconcile the existing API and the JLD-specific calls.
> 
> Since we ditched backwards compatibility, all of the `jld` prefixes should get removed as this is now _the_ API.
> 
> We should probably have a ticket opened for that and get it into alpha 1 so we don't break API after release.

Created a ticket here: https://github.com/fluree/db/issues/293

**Pull request merged**